kube-ovn-controller crash when enable ecmp

# Expected Behavior

kube-ovn-controller works fine when enable ecmp.

# Actual Behavior

kube-ovn-controller sometimes crash.

# Steps to Reproduce the Problem

1. Edit kube-ovn-controller args to add `--enable-ecmp=true`
2. Create a subnet with centralized gateway
```yaml
apiVersion: kubeovn.io/v1
kind: Subnet
metadata:
  name: default-ecmp-subnet
spec:
  protocol: IPv4
  cidrBlock: 192.168.0.0/24
  default: false
  gatewayNode: "kube-ovn-worker,kube-ovn-control-plane"
  gatewayType: centralized
  natOutgoing: true
  vpc: ovn-cluster
```
3. The restart time increase and with previous log here

```bash
E1124 05:30:09.277944      13 runtime.go:79] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
goroutine 1372 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x561003571f80?, 0x5610038e0fa8})
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000003ce0?})
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/runtime/runtime.go:49 +0x75
panic({0x561003571f80, 0x5610038e0fa8})
        /usr/local/Cellar/go/1.19.1/libexec/src/runtime/panic.go:884 +0x212
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).addPolicyRouteForCentralizedSubnetOnNode(0xc000a44000, {0xc0007e80f0, 0x16}, {0xc000c0b420, 0xa})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:1214 +0x919
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).handleAddNode(0xc000a44000, {0xc0007e80f0, 0x16})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:346 +0x2247
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).processNextAddNodeWorkItem.func1(0xc000a44000, {0x561003467da0?, 0xc000003ce0})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:116 +0x1a5
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).processNextAddNodeWorkItem(0xc000a44000)
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:122 +0x48
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).runAddNodeWorker(0xc000e356a0?)
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:87 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005242e0?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00084a788?, {0x5610038e7800, 0xc001038060}, 0x1, 0xc0001144e0)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x1000057b8?, 0x3b9aca00, 0x0, 0x38?, 0x56100117eb65?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:90 +0x25
created by github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).startWorkers
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/controller.go:845 +0xe6a
panic: assignment to entry in nil map [recovered]
        panic: assignment to entry in nil map

goroutine 1372 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000003ce0?})
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x561003571f80, 0x5610038e0fa8})
        /usr/local/Cellar/go/1.19.1/libexec/src/runtime/panic.go:884 +0x212
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).addPolicyRouteForCentralizedSubnetOnNode(0xc000a44000, {0xc0007e80f0, 0x16}, {0xc000c0b420, 0xa})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:1214 +0x919
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).handleAddNode(0xc000a44000, {0xc0007e80f0, 0x16})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:346 +0x2247
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).processNextAddNodeWorkItem.func1(0xc000a44000, {0x561003467da0?, 0xc000003ce0})
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:116 +0x1a5
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).processNextAddNodeWorkItem(0xc000a44000)
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:122 +0x48
github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).runAddNodeWorker(0xc000e356a0?)
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/node.go:87 +0x25
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0005242e0?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:155 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00084a788?, {0x5610038e7800, 0xc001038060}, 0x1, 0xc0001144e0)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x1000057b8?, 0x3b9aca00, 0x0, 0x38?, 0x56100117eb65?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
        /Users/oilbeater/workspace/pkg/mod/k8s.io/apimachinery@v0.24.4/pkg/util/wait/wait.go:90 +0x25
created by github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).startWorkers
        /Users/oilbeater/workspace/src/github.com/kubeovn/kube-ovn/pkg/controller/controller.go:845 +0xe6a

```

It will not always crash, maybe some time series problem. Detailed logs can be found here:

[kube-ovn-controller.log](https://github.com/kubeovn/kube-ovn/files/10081137/kube-ovn-controller.log)


# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  v1.24.7
  ```

- kube-ovn version:
  
  ```
  master
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

<!-- Any other additional information -->

--- a/pkg/controller/node.go
+++ b/pkg/controller/node.go
@@ -1211,6 +1211,9 @@ func (c *Controller) addPolicyRouteForCentralizedSubnetOnNode(nodeName, nodeIP s
 						continue
 					}
 					nextHops = append(nextHops, nextHop)
+					if nameIpMap == nil {
+						nameIpMap = make(map[string]string, 1)
+					}
 					nameIpMap[nodeName] = nextHop
 
 					if err = c.updatePolicyRouteForCentralizedSubnet(subnet.Name, cidrBlock, nextHops, nameIpMap); err != nil {

